### PR TITLE
Call elmo for inline guest

### PR DESCRIPTION
**Pull request by khanghoang**

### CODE REVIEW

File: server/buttons/middleware.js [hunk 1]
```diff
@@ -57,6 +57,54 @@ export function getButtonMiddleware({ logger = defaultLogger, getFundingEligibil
             logger.info(req, `button_render_version_${ render.version }`);
             logger.info(req, `button_params`, { params: JSON.stringify(params) });
 
+            // call inline guest experiment
+            const isCardFieldsEnabled = await getInlineGuestExperiment(
+                req,
+                getParams(params, req, res),
+            );
+
+            /*
```
Comment (bluepnume) [inline server/buttons/middleware.js]: Can we remove this?

File: server/buttons/middleware.js [hunk 2]
```diff
@@ -57,6 +57,54 @@ export function getButtonMiddleware({ logger = defaultLogger, getFundingEligibil
             logger.info(req, `button_render_version_${ render.version }`);
             logger.info(req, `button_params`, { params: JSON.stringify(params) });
 
+            // call inline guest experiment
+            const isCardFieldsEnabled = await getInlineGuestExperiment(
```
Comment (bluepnume) [inline server/buttons/middleware.js]: We should do this in the Promise.all below, in parallel with the other async calls we make.

File: server/buttons/middleware.js [hunk 3]
```diff
@@ -75,7 +123,7 @@ export function getButtonMiddleware({ logger = defaultLogger, getFundingEligibil
             }
 
             const buttonHTML = render.button.Buttons({
-                ...params, nonce: cspNonce, csp:   { nonce: cspNonce }, fundingEligibility, personalization
+                ...params, nonce: cspNonce, csp:   { nonce: cspNonce }, fundingEligibility, personalization, isCardFieldsEnabled
```
Comment (bluepnume) [inline server/buttons/middleware.js]: Do we need to pass this here? I think not right?

File: src/button/button.js [hunk 4]
```diff
@@ -104,7 +105,7 @@ export function setupButton({ fundingEligibility, buyerCountry: buyerGeoCountry,
                     });
                 }
 
-                if (isCardFields) {
+                if (isCardFields && isCardFieldsEnabled) {
```
Comment (bluepnume) [inline src/button/button.js]: Can we pass `isCardFieldsEnabled` into `isCardFields`, so we have a single function which does all of the eligibility checks?

File: src/button/button.js [hunk 5]
```diff
@@ -19,10 +19,11 @@ type ButtonOpts = {|
     fundingEligibility : FundingEligibilityType,
     buyerCountry? : ?$Values<typeof COUNTRY>,
     cspNonce? : string,
-    merchantID? : $ReadOnlyArray<string>
+    merchantID? : $ReadOnlyArray<string>,
+    isCardFieldsEnabled? : boolean
 |};
 
-export function setupButton({ fundingEligibility, buyerCountry: buyerGeoCountry, cspNonce: serverCSPNonce, merchantID: serverMerchantID } : ButtonOpts) : ZalgoPromise<void> {
+export function setupButton({ fundingEligibility, buyerCountry: buyerGeoCountry, cspNonce: serverCSPNonce, merchantID: serverMerchantID, isCardFieldsEnabled } : ButtonOpts) : ZalgoPromise<void> {
```
Comment (bluepnume) [inline src/button/button.js]: Maybe let's call this `isCardFieldsExperimentEnabled`

**Comment by khanghoang** [comment]
@bluepnume I fixed it, please help me to review it again

File: src/payment-flows/card-fields.js [hunk 6]
```diff
@@ -77,6 +74,25 @@ export function isCardFieldsEligible({ win, vault, onShippingChange, fundingSour
         return false;
     }
 
+    // if merchant opt-in inline guest, they will ALWAYS see inline guest guest
+    if (window.xprops.enableInlineGuest === true) {
+        return true;
+    }
+
+    // if merchant opt-out inline guest, they will NEVER see inline guest guest
+    if (window.xprops.enableInlineGuest === false) {
```
Comment (upigilam) [inline src/payment-flows/card-fields.js]: is this `enableInlineGuest` functionality defined completely? as per our discussion with josh, `false` may be not an acceptable value or is this temporary until ramp starts?...

Comment (khanghoang) [reply src/payment-flows/card-fields.js]: @upigilam I think Daniel has removed the file.

**Comment by khanghoang** [comment]
@upigilam please help me to review this PR again. Thanks.

**Comment by bluepnume** [comment]
Looks good to me!

**Comment by khanghoang** [comment]
@bluepnume can you help me to merge this PR, please? I don't have write permission.

**Pull request merged**